src/hist/named.py

```diff
@@ -10,7 +11,28 @@ def __init__(self, *args, **kwargs):
 
         super().__init__(*args, **kwargs)
         if "" in self.names:
-            raise Exception("Each axes in the NamedHist instance should have a name.")
+            raise Exception(
+                f"Each axes in the {self.__class__.__name__} instance should have a name"
+            )
+
+    def project(self, *args: Tuple[Union[int, str]]):
+        """
+        Projection of axis idx.
+        """
+
+        if all(isinstance(x, int) for x in args):
```

Comment:
Move this function to BaseHist and then the function here will just have:

```python

if all(isinstance(x, str) for x in args):
    return super().project(*args)
else:
    raise ...
```

Reply:
That's exactly what I do in my new commit.